sdk/program/src/alt_bn128.rs

```diff
@@ -245,14 +269,13 @@ mod target_arch {
                 .map_err(AltBn128Error::TryIntoVecError)?,
             )
             .try_into()?;
-            vec_pairs.push((g1.into(), g2.into()));
+            vec_b.push(g2);
         }
 
         let mut result = BigInteger256::from(0u64);
-        let res =
-            <Bn<Parameters> as PairingEngine>::product_of_pairings((vec_pairs[..ele_len]).iter());
-        type GT = <ark_ec::models::bn::Bn<ark_bn254::Parameters> as ark_ec::PairingEngine>::Fqk;
-        if res == GT::one() {
+        let res = <Bn<Config> as Pairing>::multi_pairing(vec_a.iter(), vec_b.iter());
```

Comment:
I would prefer two `.iter().map(|pair| pair.x)`s instead of two `Vec`s to reduce the workload of the memory allocator.